code/modules/power/battery_ion.dm

```diff
@@ -0,0 +1,70 @@
+// Generic battery machine
+// stores power
+
+/obj/machinery/power/battery/portable
+	name = "portable power storage unit"
+	desc = "A IOn-model portable storage unit, used to transport charge around the station."
+	icon_state = "port_smes"
+	density = 1
+	anchored = 0
+	use_power = 0
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE
+
+	var/obj/machinery/power/battery_port/connected_to
+
```

Comment:
You need a destroy proc to remove references here

Destroy()
if(connected_to)
connected_to.connected = null
connected_to = null